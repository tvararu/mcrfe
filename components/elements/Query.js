import { Query } from "react-apollo";
import Loading from "./Loading";

export default ({ query, variables, children }) => (
  <Query query={query} variables={variables}>
    {({ loading, error, data }) => {
      if (error) return <pre>Error: {JSON.stringify(error, null, 2)}</pre>;
      if (loading) return <Loading />;

      return children({ data });
    }}
  </Query>
);
