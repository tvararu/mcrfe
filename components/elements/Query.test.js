import renderer from "react-test-renderer";
import Query from "./Query";

let mockedPayload = {};

jest.mock("react-apollo", () => ({
  Query: ({ children }) => children(mockedPayload)
}));

describe("Query", () => {
  [
    { error: "Something went wrong" },
    { loading: true },
    { data: "Lorem" }
  ].forEach(payload => {
    it(`renders correctly with ${Object.keys(payload).pop()}`, () => {
      mockedPayload = payload;
      const component = renderer.create(
        <Query>{({ data }) => <p>{data}</p>}</Query>
      );
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
