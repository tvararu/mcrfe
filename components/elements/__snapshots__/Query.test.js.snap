// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query renders correctly with data 1`] = `
<p>
  Lorem
</p>
`;

exports[`Query renders correctly with error 1`] = `
<pre>
  Error: 
  "Something went wrong"
</pre>
`;

exports[`Query renders correctly with loading 1`] = `
<p
  className="jsx-3874535168 govuk-body"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <svg
    className="jsx-3874535168 full-crown"
    focusable="false"
    height="32"
    role="presentation"
    style={
      Object {
        "margin": "-6px 0",
        "marginRight": "5px",
      }
    }
    viewBox="0 0 132 97"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="jsx-3874535168"
      d="M25 30.2c3.5 1.5 7.7-.2 9.1-3.7 1.5-3.6-.2-7.8-3.9-9.2-3.6-1.4-7.6.3-9.1 3.9-1.4 3.5.3 7.5 3.9 9zM9 39.5c3.6 1.5 7.8-.2 9.2-3.7 1.5-3.6-.2-7.8-3.9-9.1-3.6-1.5-7.6.2-9.1 3.8-1.4 3.5.3 7.5 3.8 9zM4.4 57.2c3.5 1.5 7.7-.2 9.1-3.8 1.5-3.6-.2-7.7-3.9-9.1-3.5-1.5-7.6.3-9.1 3.8-1.4 3.5.3 7.6 3.9 9.1zm38.3-21.4c3.5 1.5 7.7-.2 9.1-3.8 1.5-3.6-.2-7.7-3.9-9.1-3.6-1.5-7.6.3-9.1 3.8-1.3 3.6.4 7.7 3.9 9.1zm64.4-5.6c-3.6 1.5-7.8-.2-9.1-3.7-1.5-3.6.2-7.8 3.8-9.2 3.6-1.4 7.7.3 9.2 3.9 1.3 3.5-.4 7.5-3.9 9zm15.9 9.3c-3.6 1.5-7.7-.2-9.1-3.7-1.5-3.6.2-7.8 3.7-9.1 3.6-1.5 7.7.2 9.2 3.8 1.5 3.5-.3 7.5-3.8 9zm4.7 17.7c-3.6 1.5-7.8-.2-9.2-3.8-1.5-3.6.2-7.7 3.9-9.1 3.6-1.5 7.7.3 9.2 3.8 1.3 3.5-.4 7.6-3.9 9.1zM89.3 35.8c-3.6 1.5-7.8-.2-9.2-3.8-1.4-3.6.2-7.7 3.9-9.1 3.6-1.5 7.7.3 9.2 3.8 1.4 3.6-.3 7.7-3.9 9.1zM69.7 17.7l8.9 4.7V9.3l-8.9 2.8c-.2-.3-.5-.6-.9-.9L72.4 0H59.6l3.5 11.2c-.3.3-.6.5-.9.9l-8.8-2.8v13.1l8.8-4.7c.3.3.6.7.9.9l-5 15.4v.1c-.2.8-.4 1.6-.4 2.4 0 4.1 3.1 7.5 7 8.1h.2c.3 0 .7.1 1 .1.4 0 .7 0 1-.1h.2c4-.6 7.1-4.1 7.1-8.1 0-.8-.1-1.7-.4-2.4V34l-5.1-15.4c.4-.2.7-.6 1-.9zM66 92.8c16.9 0 32.8 1.1 47.1 3.2 4-16.9 8.9-26.7 14-33.5l-9.6-3.4c1 4.9 1.1 7.2 0 10.2-1.5-1.4-3-4.3-4.2-8.7L108.6 76c2.8-2 5-3.2 7.5-3.3-4.4 9.4-10 11.9-13.6 11.2-4.3-.8-6.3-4.6-5.6-7.9 1-4.7 5.7-5.9 8-.5 4.3-8.7-3-11.4-7.6-8.8 7.1-7.2 7.9-13.5 2.1-21.1-8 6.1-8.1 12.3-4.5 20.8-4.7-5.4-12.1-2.5-9.5 6.2 3.4-5.2 7.9-2 7.2 3.1-.6 4.3-6.4 7.8-13.5 7.2-10.3-.9-10.9-8-11.2-13.8 2.5-.5 7.1 1.8 11 7.3L80.2 60c-4.1 4.4-8 5.3-12.3 5.4 1.4-4.4 8-11.6 8-11.6H55.5s6.4 7.2 7.9 11.6c-4.2-.1-8-1-12.3-5.4l1.4 16.4c3.9-5.5 8.5-7.7 10.9-7.3-.3 5.8-.9 12.8-11.1 13.8-7.2.6-12.9-2.9-13.5-7.2-.7-5 3.8-8.3 7.1-3.1 2.7-8.7-4.6-11.6-9.4-6.2 3.7-8.5 3.6-14.7-4.6-20.8-5.8 7.6-5 13.9 2.2 21.1-4.7-2.6-11.9.1-7.7 8.8 2.3-5.5 7.1-4.2 8.1.5.7 3.3-1.3 7.1-5.7 7.9-3.5.7-9-1.8-13.5-11.2 2.5.1 4.7 1.3 7.5 3.3l-4.7-15.4c-1.2 4.4-2.7 7.2-4.3 8.7-1.1-3-.9-5.3 0-10.2l-9.5 3.4c5 6.9 9.9 16.7 14 33.5 14.8-2.1 30.8-3.2 47.7-3.2z"
      fillRule="evenodd"
    />
  </svg>
  <svg
    className="jsx-3874535168 empty-crown"
    focusable="false"
    height="32"
    role="presentation"
    style={
      Object {
        "margin": "-6px 0",
        "marginRight": "5px",
      }
    }
    viewBox="0 0 132 97"
    width="36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      className="jsx-3874535168"
      d="M25,30.2 M69.7,17.7 L69.7,17.7 L78.6,22.4 L78.6,9.3 L69.7,12.1 L69.7,12.1 C69.5,11.8 69.2,11.5 68.8,11.2 L68.8,11.2 L72.4,0 L66,0 L59.6,0 L63.1,11.2 L63.1,11.2 C62.8,11.5 62.5,11.7 62.2,12.1 L62.2,12.1 L53.4,9.3 L53.4,22.4 L62.2,17.7 L62.2,17.7 C62.5,18 62.8,18.4 63.1,18.6 L58.1,34 C58.1,34 58.1,34 58.1,34.1 L58.1,34.1 L58.1,34.1 C57.9,34.9 57.7,35.7 57.7,36.5 C57.7,40.6 60.8,44 64.7,44.6 L64.9,44.6 C65.2,44.6 65.6,44.7 65.9,44.7 C66.3,44.7 66.6,44.7 66.9,44.6 L67.1,44.6 C71.1,44 74.2,40.5 74.2,36.5 C74.2,35.7 74.1,34.8 73.8,34.1 L73.8,34.1 L73.8,34.1 C73.8,34.1 73.8,34.1 73.8,34 L68.7,18.6 C69.1,18.4 69.4,18 69.7,17.7 L69.7,17.7 M66,92.8 C82.9,92.8 98.8,93.9 113.1,96 C117.1,79.1 122,69.3 127.1,62.5 L117.5,59.1 C118.5,64 118.6,66.3 117.5,69.3 C116,67.9 114.5,65 113.3,60.6 L108.6,76 C111.4,74 113.6,72.8 116.1,72.7 C111.7,82.1 106.1,84.6 102.5,83.9 C98.2,83.1 96.2,79.3 96.9,76 C97.9,71.3 102.6,70.1 104.9,75.5 C109.2,66.8 101.9,64.1 97.3,66.7 C104.4,59.5 105.2,53.2 99.4,45.6 C91.4,51.7 91.3,57.9 94.9,66.4 C90.2,61 82.8,63.9 85.4,72.6 C88.8,67.4 93.3,70.6 92.6,75.7 C92,80 86.2,83.5 79.1,82.9 C68.8,82 68.2,74.9 67.9,69.1 C70.4,68.6 75,70.9 78.9,76.4 L80.2,60 C76.1,64.4 72.2,65.3 67.9,65.4 C69.3,61 75.9,53.8 75.9,53.8 L66.4,53.8 L66.4,53.8 L66.4,53.8 L55.5,53.8 C55.5,53.8 61.9,61 63.4,65.4 C59.2,65.3 55.4,64.4 51.1,60 L52.5,76.4 C56.4,70.9 61,68.7 63.4,69.1 C63.1,74.9 62.5,81.9 52.3,82.9 C45.1,83.5 39.4,80 38.8,75.7 C38.1,70.7 42.6,67.4 45.9,72.6 C48.6,63.9 41.3,61 36.5,66.4 C40.2,57.9 40.1,51.7 31.9,45.6 C26.1,53.2 26.9,59.5 34.1,66.7 C29.4,64.1 22.2,66.8 26.4,75.5 C28.7,70 33.5,71.3 34.5,76 C35.2,79.3 33.2,83.1 28.8,83.9 C25.3,84.6 19.8,82.1 15.3,72.7 C17.8,72.8 20,74 22.8,76 L18.1,60.6 C16.9,65 15.4,67.8 13.8,69.3 C12.7,66.3 12.9,64 13.8,59.1 L4.3,62.5 C9.3,69.4 14.2,79.2 18.3,96 C33.1,93.9 49.1,92.8 66,92.8 L66,92.8  Z "
      fillRule="evenodd"
    />
  </svg>
  Loading
  
</p>
`;
